wcfsetup/install/files/lib/data/user/UserAction.class.php

```diff
@@ -1004,12 +1061,13 @@ public function validateResendActivationMail() {
 			throw new PermissionDeniedException();
 		}
 		
-		if (REGISTER_ACTIVATION_METHOD != 1) {
+		if (!(REGISTER_ACTIVATION_METHOD & User::REGISTER_ACTIVATION_USER)) {
 			throw new IllegalLinkException();
 		}  
 		
 		foreach ($this->objects as $object) {
-			if (!$object->activationCode) {
+			/** @var UserEditor $object */
+			if (!$object->requiresEmailActivation()) {
```

Comment:
Should this rather be `canEmailConfirm`?

Reply:
Yeah, that would actually make more sense.